python/hail/dataset.py

```diff
@@ -750,6 +750,63 @@ def annotate_variants_expr(self, condition):
         pargs = ['annotatevariants', 'expr', '-c', condition]
         return self.hc.run_command(self, pargs)
 
+    def annotate_variants_keytable(self, keytable, condition, vdsKey=None):
```

Comment:
I think this is more consistent with the KeyTable API.  The key is not a structure, it is an (ordered) list of toplevel columns.  This was done explicitly to have a reliable way to predict the ordering of fields in the key structure.  Runtime values of type key signature should only be realized to perform joins.